src/lib/hooks/use-on-click-outside.js

```diff
@@ -0,0 +1,29 @@
+import { useEffect } from 'react'
```

Comment:
Added some hooks as used in `popover`. Added to `/lib` as it feels like they may be useful elsewhere as well.

Reply:
This is a really neat hook! Seems like it would be useful [in `ProductSwitcher`](https://github.com/hashicorp/dev-portal/blob/assembly-ui-v1/src/components/product-switcher/index.tsx#L67-L76) as well, what do you think? Not asking you to implement there, mainly asking to make sure I understand the code here fully. 😊

Reply:
@zchsh I was wondering because I think we might want to name this `useClickOutside` instead of including `Dropdown` in the name, but that's a _nit_ and I don't feel super strongly about needing to rename now

Reply:
Great call, have renamed - I think the name was just a holdout from lifting this function out of a context where it was for a dropdown, even though none of the code is really dropdown specific.

Which probably answers the `ProductSwitcher` question as well - that seems like a great use case!